src/modules/logicModule/BountyManager.sol

```diff
@@ -0,0 +1,528 @@
+//At the current state of the project the IRoleAuthorizer isnt the standardized Interface to use,
+//That means there is a possibility function calls will fail if the roleAuthorizer isnt implemented as the Authorizer
+//This will be changed in the coming future,
+//but till then the RoleAuthorizer has to be selected as the Authorizer Module of the proposal if the BountyManager is to be used
+
+contract BountyManager is IBountyManager, Module, PaymentClient {
+    using SafeERC20 for IERC20;
+    using EnumerableSet for EnumerableSet.UintSet;
+    using LinkedIdList for LinkedIdList.List;
+
+    //--------------------------------------------------------------------------
+    // Modifiers
+
+    //@todo Reminder that this will be moved into the Module Contract at a later point of time
+    modifier onlyRole(uint8 roleId) {
+        if (
+            !IRoleAuthorizer(address(__Module_proposal.authorizer()))
+                .isAuthorized(roleId, _msgSender())
+        ) {
+            revert Module__BountyManager__OnlyRole(roleId, address(this));
+        }
+        _;
+    }
+
+    modifier onlyClaimContributor(uint claimId) {
+        address sender = _msgSender();
+        Contributor[] memory contribs = _claimRegistry[claimId].contributors;
+        uint length = contribs.length;
+        bool found;
+        for (uint i; i < length; i++) {
+            if (contribs[i].addr == sender) {
+                found = true;
+                break;
+            }
+        }
+
+        if (!found) {
+            revert Module__BountyManager__OnlyClaimContributor();
+        }
+        _;
+    }
+
+    modifier validPayoutAmounts(
+        uint minimumPayoutAmount,
+        uint maximumPayoutAmount
+    ) {
+        if (
+            minimumPayoutAmount == 0
+                || maximumPayoutAmount < minimumPayoutAmount
+        ) {
+            revert Module__BountyManager__InvalidPayoutAmounts();
+        }
+        _;
+    }
+
+    modifier validBountyId(uint bountyId) {
+        if (!isExistingBountyId(bountyId)) {
+            revert Module__BountyManager__InvalidBountyId();
+        }
+        _;
+    }
+
+    modifier validClaimId(uint claimId) {
+        if (!isExistingClaimId(claimId)) {
+            revert Module__BountyManager__InvalidClaimId();
+        }
+        _;
+    }
+
+    function validContributorsForBounty(
+        Contributor[] memory contributors,
+        Bounty memory bounty
+    ) internal view {
+        //@update to be in correct range
+        uint length = contributors.length;
+        //length cant be zero
+        if (length == 0) {
+            revert Module__BountyManager__InvalidContributorsLength();
+        }
+        uint totalAmount;
+        uint currentAmount;
+        address contrib;
+        for (uint i; i < length; i++) {
+            currentAmount = contributors[i].claimAmount;
+
+            //amount cant be zero
+            if (currentAmount == 0) {
```

Comment:
I don't really think having a zero payout for one of the contributors should be considered as an erroring state. I think it's perfectly reasonable to assume that 1 out of 100 contributor could not commit some time to the bounty and therefore their payout turned out to be 0...

Let me know if I'm making some wrong assumptions.

Reply:
Why would you include a contributor qho isnt get paid anything?
I think this is more of a design decision, there are points for both sides
If at some point there is a cry for a change in production we can still change it, but until then i would keep it